src/main/java/org/wildfly/security/auth/provider/FileSystemSecurityRealm.java

```diff
@@ -67,8 +68,11 @@
 import org.wildfly.security.auth.server.NameRewriter;
 import org.wildfly.security.password.Password;
 import org.wildfly.security.password.PasswordFactory;
-import org.wildfly.security.password.PasswordUtil;
+import org.wildfly.security.password.util.ModularCrypt;
 import org.wildfly.security.password.interfaces.ClearPassword;
+import org.wildfly.security.password.spec.ClearPasswordSpec;
+import org.wildfly.security.password.spec.DefaultPasswordSpecEncoding;
+import org.wildfly.security.password.spec.PasswordSpec;
 import org.wildfly.security.util.ByteIterator;
 import org.wildfly.security.util.CodePointIterator;
 
```

Comment:
The main change here is about how to write the identity. So we could reuse how the top and bottom level elements are created and customize only the actual body.

This is the case where you are creating or updating the identity.

I can also revert that, if you want.